code/game/machinery/doors/door.dm

```diff
@@ -313,8 +316,51 @@
 
 /obj/machinery/door/update_icon_state()
 	icon_state = "[base_icon_state][density]"
+
+	if(hasPower() && unres_sides)
+		set_light(2, 1)
+	else
+		set_light(0)
+
 	return ..()
 
+/obj/machinery/door/update_overlays()
+	. = ..()
+
+	if(hasPower() && unres_sides)
+		if(unres_sides & NORTH)
+			var/image/I = image(icon='icons/obj/doors/airlocks/station/overlays.dmi', icon_state="unres_n")
+			switch(dir)
+				if(NORTH)
+					I.pixel_y = 31
+				if(SOUTH)
+					I.pixel_y = 6
+			. += I
+		if(unres_sides & SOUTH)
+			var/image/I = image(icon='icons/obj/doors/airlocks/station/overlays.dmi', icon_state="unres_s")
+			switch(dir)
+				if(NORTH)
+					I.pixel_y = -6
+				if(SOUTH)
+					I.pixel_y = -31
+			. += I
+		if(unres_sides & EAST)
+			var/image/I = image(icon='icons/obj/doors/airlocks/station/overlays.dmi', icon_state="unres_e")
+			switch(dir)
+				if(EAST)
+					I.pixel_x = 31
+				if(WEST)
+					I.pixel_x = 6
+			. += I
+		if(unres_sides & WEST)
+			var/image/I = image(icon='icons/obj/doors/airlocks/station/overlays.dmi', icon_state="unres_w")
+			switch(dir)
+				if(EAST)
+					I.pixel_x = -6
+				if(WEST)
+					I.pixel_x = -31
+			. += I
```

Comment:
I feel like this whole thing could probably be condensed down to two if statements

Reply:
Well let me know if you figure out how.............